networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/realization.py

```diff
@@ -0,0 +1,273 @@
+import datetime
+import itertools
+import json
+import time
+
+from networking_nsxv3.common.locking import LockManager
+from oslo_config import cfg
+from oslo_log import log as logging
+
+LOG = logging.getLogger(__name__)
+
+
+class AgentRealizer(object):
+
+    def __init__(self, rpc, callback, kpi, provider, legacy_provider):
+        self.rpc = rpc
+        self.callback = callback
+        self.kpi = kpi
+        self.provider = provider
+        self.legacy_provider = legacy_provider
+        self.age = int(time.time())
+        # Initializing metadata
+        self.all(dryrun=True)
+
+    def _os_meta(self, query):
+        step = cfg.CONF.AGENT.rpc_max_records_per_query
+        cursor = 0
+        meta = dict()
+        while cursor != -1:
+            result = query(step, cursor)
+            meta.update({k:v for (k,v) in result})
```

Comment:
this implies that result entities consists of a tuple of 2 elements, but for example `get_ports_with_revisions` return a list of 3-tuples, causing this error: 
```python
Traceback (most recent call last):
  File "/Users/d072895/Library/Application Support/JetBrains/Toolbox/apps/PyCharm-P/ch-0/211.7142.13/PyCharm.app/Contents/plugins/python/helpers/pydev/pydevd.py", line 1491, in _exec
    func()
  File "/Users/d072895/Workspace/queens/networking-nsx-t/networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/agent.py", line 354, in main
    NSXv3Manager(rpc=nsxv3_rpc.NSXv3ServerRpcApi()),
  File "/Users/d072895/Workspace/queens/networking-nsx-t/networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/agent.py", line 116, in __init__
    legacy_provider=legacy_provider)
  File "/Users/d072895/Workspace/queens/networking-nsx-t/networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/realization.py", line 23, in __init__
    self.all(dryrun=True)
  File "/Users/d072895/Workspace/queens/networking-nsx-t/networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/realization.py", line 65, in all
    port_meta = self._os_meta(r.get_ports_with_revisions)
  File "/Users/d072895/Workspace/queens/networking-nsx-t/networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/realization.py", line 31, in _os_meta
    meta.update({k:v for (k,v) in result})
  File "/Users/d072895/Workspace/queens/networking-nsx-t/networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/realization.py", line 31, in <dictcomp>
    meta.update({k:v for (k,v) in result})
ValueError: too many values to unpack
```

Reply:
![Screenshot 2021-05-21 at 14 05 13](https://user-images.githubusercontent.com/2862494/119134587-9f1a3b80-ba3d-11eb-97b1-05c29ff5ebd2.png)